GameOfLife/Logic/GameStateChecker.cs

```diff
@@ -0,0 +1,44 @@
+﻿namespace GameOfLife
+{
+    /// <summary>
+    /// Checks variable values for curent game state
+    /// </summary>
+    public class GameStateChecker
```

Comment:
Why did you choose to create a separate class for getting these values instead of adding these methods in `GameFieldData` class? Since `GameFieldData` class is responsible about the, you know, game field, it would only make sense that it also knows how many live cells are in the field and it could also track its own iterations.